src/metabase/permissions_rest/data_permissions/graph.clj

```diff
@@ -161,12 +162,26 @@
         [:int {:title "table-id" :min 0}]
         ::permissions.schema/data-permission-value]]]]]])
 
+(defn- error-on-conflicting-entries [data-perms]
+  (let [grouped (group-by (fn [perm]
+                            [(:group-id perm) (:db-id perm) (:type perm)])
+                          data-perms)]
+    (run!
+     (fn [[[group-id db-id perm-type] perms]]
+       (let [db-level (first (filter (comp nil? :table-id) perms))
+             individual (filter :table-id perms)]
+         (when (and db-level (seq individual))
+           (log/errorf "Conflicting entries for db_id:%s, group_id:%s, perm_type:%s, IDS: db-level:%s table-level:[%s]" db-id group-id (symbol perm-type)
```

Comment:
Let's have fixed log messages but add the structured data we need, something like:

```
(log/with-context {:db_id db-id :table_id table-id :perm_type perm-type} (log/errorf "Conflicting entries for database and tables")
```